upup/pkg/fi/cloudup/awstasks/vpc_dhcpoptions_association.go

```diff
@@ -82,17 +82,83 @@ func (s *VPCDHCPOptionsAssociation) CheckChanges(a, e, changes *VPCDHCPOptionsAs
 }
 
 func (_ *VPCDHCPOptionsAssociation) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *VPCDHCPOptionsAssociation) error {
-	if changes.DHCPOptions != nil {
-		glog.V(2).Infof("calling EC2 AssociateDhcpOptions")
-		request := &ec2.AssociateDhcpOptionsInput{
-			VpcId:         e.VPC.ID,
-			DhcpOptionsId: e.DHCPOptions.ID,
-		}
+	if changes.DHCPOptions == nil {
+		return nil
+	}
+	glog.V(2).Infof("calling EC2 AssociateDhcpOptions")
+	request := &ec2.AssociateDhcpOptionsInput{
+		VpcId:         e.VPC.ID,
+		DhcpOptionsId: e.DHCPOptions.ID,
+	}
 
-		_, err := t.Cloud.EC2().AssociateDhcpOptions(request)
-		if err != nil {
-			return fmt.Errorf("error creating VPCDHCPOptionsAssociation: %v", err)
+	_, err := t.Cloud.EC2().AssociateDhcpOptions(request)
+	if err != nil {
+		return fmt.Errorf("error creating VPCDHCPOptionsAssociation: %v", err)
+	}
+
+	// This part is a little annoying. If you're running in a region
+	// with where there is no default-looking DHCP option set, when
+	// you create any VPC, AWS will create a default-looking DHCP
+	// option set for you. If you then re-associate (as below) or
+	// delete the VPC, the option set will hang around. However, if
+	// you have a default-looking DHCP option set (for example, an
+	// unmodified default VPC) and create a VPC, AWS will associate
+	// the VPC with the DHCP option set of the default VPC. There's no
+	// signal as to whether the option set returned is the default or
+	// was created along with the VPC.
+	//
+	// Solution: When we reassociate the DHCP option set, try a
+	// courtesy delete on it. If that gets a DependencyViolation,
+	// it's still in use and we move on.
+	if *a.DHCPOptions.ID == "default" {
```

Comment:
At least in theory, a could be nil here.  I'm not sure it really can though, but I suggest we guard against it here (`if a== nil || *a.DHCPOptions.ID == "default"`)